Add Respoke-SDK header to ws and http requests

chadxz opened the pull request:
This patch adds the "Respoke-SDK" header to websocket connect, requests, and HTTP[S] requests. This will help the Respoke team understand SDK usage information. It contains the name of the SDK
and the os name and version.

I wasn't able to pull the SDK version at this time, because I don't think there's any way to pull the version out of the cocoapods .podspec file, since it is essentially ruby source. I am thinking we'll need to pull the version out into a separate file, and read that file in the .podspec as well as our source. I'm not sure the best solution to that, though, so I would appreciate any suggestions.

<img width="547" alt="screen shot 2015-09-27 at 12 37 59 pm" src="https://cloud.githubusercontent.com/assets/309219/10124198/e53815a8-6514-11e5-8887-993d3b1615a0.png">
<img width="1001" alt="screen shot 2015-09-27 at 11 59 29 am" src="https://cloud.githubusercontent.com/assets/309219/10124197/e537d5b6-6514-11e5-9923-565a33938093.png">
<img width="606" alt="screen shot 2015-09-27 at 11 59 54 am" src="https://cloud.githubusercontent.com/assets/309219/10124199/e538b1c0-6514-11e5-8297-3963cef46825.png">

Related to MER-4340

P.S. I tested this in the Simulator and most all the tests passed, except the ones that require WebRTC which isn't available in the Simulator.

<img width="253" alt="screen shot 2015-09-27 at 12 05 23 pm" src="https://cloud.githubusercontent.com/assets/309219/10124200/e5395698-6514-11e5-9dea-c601cd9c3b7e.png">

Commit:
Add Respoke-SDK header to ws and http requests

This patch adds the "Respoke-SDK" header to websocket connect,
requests, and HTTP[S] requests. This will help the Respoke team
understand SDK usage information. It contains the name of the SDK
and the os name and version.

Related to MER-4340

chadxz commented:
@leedm777 pls to review? :smirk:

leedm777 commented on RespokeSDK/NSString+urlencode.m:
```diff
@@ -0,0 +1,39 @@
+
+@implementation NSString (NSString_Extended)
+
+- (NSString *)urlencode {
```
This isn't already in the iOS SDK? You should probably cite where you got the code in a comment here.

chadxz replied:
everything I can tell from reading on the internet is that it doesn't do it _properly_. It misses some important characters like "&" and "/". I'll add a comment about the source of the code. All it does is encode everything that isn't in the small subset of allowed characters `[a-zA-z0-9\.\-_~]` and it replaces ' ' (space) with '+'. 
http://stackoverflow.com/a/8088484/355743

leedm777 commented:
@chadxz Objective-C still looks like gibberish to me. But this gibberish looks right. I guess.

Commit:
Add comment about source of urlencode category

chadxz commented:
I was able to get a temporary measure in place for displaying the sdk version. I simply created a file that exposes the SDK version as a function you can call, similar to the way we do chan_respoke versioning. Only problem is that this file isn't generated by a build script, so when we want to release a new version, we'll need to not only update the .podspec file, but also this version file. It's good enough for now, but if we ever start iterating on this sdk more, we'll probably want to improve it with some automation.

new changes in action: 

<img width="577" alt="screen shot 2015-09-28 at 9 59 04 pm" src="https://cloud.githubusercontent.com/assets/309219/10154406/80ffc446-662c-11e5-91cf-b57dc49ae7e2.png">

Commit:
Add RespokeVersion.h/.m for rudimentary versioning

This patch adds a very simplistic method of obtaining the version
information from inside the SDK. This was needed for proper version
reporting in the sdk header. In the future we'll probably want to
automate the updating of this file or implement some alternative
method of allowing the SDK access to its version, but for now this
works.
